src/java/org/apache/cassandra/index/sai/disk/v1/V1SearchableIndex.java

```diff
@@ -169,6 +175,12 @@ public List<RangeIterator<Long>> search(Expression expression,
     {
         List<RangeIterator<Long>> iterators = new ArrayList<>();
 
+        if(keyRange instanceof Bounds && keyRange.left == keyRange.right && keyRange.left instanceof DecoratedKey)
+        {
+            if (!filter.isPresent((DecoratedKey)keyRange.left))
+                return iterators;
+        }
```

Comment:
Nit: I checked out the PR and ran the test, but I didn't see the stack enter this conditional when running with a debugger. Is that expected? I assumed the test could hit this part of the PR.